Task linkedlist reversed

zuwarka opened the pull request.

zuwarka committed:
Update LinkedList.rb

Добавлены методы reverse и reverse!

zuwarka committed:
Update Node.rb

Добавлены методы reverse_node и reverse_node! для рекурсивной работы таких же методов в Списке.

zuwarka committed:
Update Node.rb

Отредактирован конструктор

Alex6Alex commented on homework05 (18.07.2022)/LinkedList.rb:
```diff
@@ -88,5 +72,37 @@ def to_s
     end
     print ")"
+  end
+      # -------------------------------------------------- Добавлено от 12.08.22
```
Такой комментарий можно не делать, при необходимости всегда можно узнать кто и когда добавлял код

Alex6Alex commented on homework05 (18.07.2022)/LinkedList.rb:
```diff
@@ -88,5 +72,37 @@ def to_s
     print ")"
+  end
+      # -------------------------------------------------- Добавлено от 12.08.22
+  def reverse(node = @head, previous = nil)
```
Это аргументы хорошо бы убрать, чтобы данный метод не позволял что-то ему передать

Alex6Alex commented on homework05 (18.07.2022)/LinkedList.rb:
```diff
@@ -88,5 +72,37 @@ def to_s
+    return unless node
+    return node unless node.next_node
+
+    to_s(node.reverse_node)
```
Цель задачи была не в отображении элементов списка, а в возврате из метода нового объекта класса LinkedList

Alex6Alex commented on homework05 (18.07.2022)/LinkedList.rb:
```diff
@@ -88,5 +72,37 @@ def to_s
+    return unless @head
+    return @head unless @head.next_node
+
+    to_s(@head.reverse_node!)
```
А тут вернуть сам текущий объект

Alex6Alex commented on homework05 (18.07.2022)/Node.rb:
```diff
@@ -1,11 +1,28 @@
   def to_s
     @data.to_s
   end
+  # -------------------------------------------------- Добавлено от 12.08.22
```
Лишний комментарий

zuwarka committed:
Update LinkedList.rb

Добавлена передача аргумента в конструктор, исправлены методы reverse (возвращение нового объекта LinkedList) и reverse! (возвращение того же объекта)
